share/eds/store_test.go

```diff
@@ -0,0 +1,170 @@
+package eds
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/filecoin-project/dagstore/shard"
+	"github.com/ipfs/go-datastore"
+	ds_sync "github.com/ipfs/go-datastore/sync"
+	"github.com/ipld/go-car"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/share"
+
+	"github.com/celestiaorg/celestia-app/pkg/da"
+	"github.com/celestiaorg/rsmt2d"
+)
+
+// TestEDSStore_PutRegistersShard tests if Put registers the shard on the underlying DAGStore
+func TestEDSStore_PutRegistersShard(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	edsStore, err := newStore(t)
+	require.NoError(t, err)
+	err = edsStore.Start(ctx)
+	require.NoError(t, err)
+
+	eds, dah := randomEDS(t)
+
+	// shard hasn't been registered yet
+	has, err := edsStore.Has(ctx, dah)
+	assert.False(t, has)
+	assert.Error(t, err, "shard not found")
+
+	err = edsStore.Put(ctx, dah, eds)
+	assert.NoError(t, err)
+
+	_, err = edsStore.dgstr.GetShardInfo(shard.KeyFromString(dah.String()))
```

Comment:
we should also be verifying that the information we is as expected, not just that it returns without error.

Reply:
The integrity after Putting and Getting is tested through the GetCAR tests, this test is only responsible for ensuring that a Put registers the shard key on the dagstore